examples/reshare/src/dkg/actor.rs

```diff
@@ -140,551 +151,445 @@ where
+                        // The >= check handles restart after midpoint acknowledgment.
+                        if relative_height >= mid_point {
+                            if let Some(ref mut ds) = dealer_state {
+                                ds.finalize();
                             }
+                        }
 
-                            let self_idx = all_participants
-                                .position(&self.signer.public_key())
-                                .expect("self must be a participant");
-                            info!(
-                                participant = self_idx,
-                                "dkg completed successfully, persisted outcome."
-                            );
+                        // Continue if not the last block in the epoch
+                        if is_last_block_in_epoch(BLOCKS_PER_EPOCH, block.height).is_none() {
+                            // Acknowledge block processing
+                            response.acknowledge();
+                            continue;
+                        }
 
-                            break;
+                        // Finalize the round before acknowledging
+                        let logs = storage.logs(epoch);
+                        let (success, next_round, next_output, next_share) =
+                            if let Some(ps) = player_state.take() {
+                                match ps.finalize(logs, 1) {
+                                    Ok((new_output, new_share)) => (
+                                        true,
+                                        epoch_state.round + 1,
+                                        Some(new_output),
+                                        Some(new_share),
+                                    ),
+                                    Err(_) => (
+                                        false,
+                                        epoch_state.round,
+                                        epoch_state.output.clone(),
+                                        epoch_state.share.clone(),
+                                    ),
+                                }
+                            } else {
+                                match observe(round.clone(), logs, 1) {
+                                    Ok(output) => (true, epoch_state.round + 1, Some(output), None),
+                                    Err(_) => (
+                                        false,
+                                        epoch_state.round,
+                                        epoch_state.output.clone(),
+                                        epoch_state.share.clone(),
+                                    ),
+                                }
+                            };
+                        if success {
+                            info!(?epoch, "epoch succeeded");
+                        } else {
+                            warn!(?epoch, "epoch failed");
```

Comment:
nit: These logs aren't very clear - the epoch failing sounds like a consensus/orchestrator issue. Maybe:
```suggestion
                        if success {
                            info!(?epoch, "reshare succeeded");
                        } else {
                            warn!(?epoch, "reshare failed");
```